pydantic/_internal/_dataclasses.py

```diff
@@ -55,57 +53,30 @@ def set_dataclass_fields(cls: type[StandardDataclass], types_namespace: dict[str
 def complete_dataclass(
     cls: type[Any],
     config_wrapper: ConfigWrapper,
-    *,
-    raise_errors: bool = True,
-    types_namespace: dict[str, Any] | None = None,
-) -> bool:
+) -> None:
     """
     Prepare a raw class to become a pydantic dataclass.
 
-    Returns `True` if the validation construction is successfully completed, else `False`.
-
     This logic is called on a class which is yet to be wrapped in `dataclasses.dataclass()`.
     """
-    cls_name = cls.__name__
-
     if hasattr(cls, '__post_init_post_parse__'):
         warnings.warn(
             'Support for `__post_init_post_parse__` has been dropped, the method will not be called', DeprecationWarning
         )
 
-    types_namespace = _typing_extra.get_cls_types_namespace(cls, types_namespace)
+    types_namespace = _typing_extra.get_cls_types_namespace(cls)
     typevars_map = get_standard_typevars_map(cls)
     gen_schema = GenerateSchema(
         config_wrapper,
         types_namespace,
         typevars_map,
     )
 
-    try:
-        get_core_schema = getattr(cls, '__get_pydantic_core_schema__', None)
-        if get_core_schema:
-            schema = get_core_schema(cls, partial(gen_schema.generate_schema, from_dunder_get_core_schema=False))
-        else:
-            schema = gen_schema.generate_schema(cls, False)
-    except PydanticUndefinedAnnotation as e:
-        if raise_errors:
-            raise
-        if config_wrapper.undefined_types_warning:
-            config_warning_string = (
-                f'`{cls_name}` has an undefined annotation: `{e.name}`. '
-                f'It may be possible to resolve this by setting '
-                f'undefined_types_warning=False in the config for `{cls_name}`.'
-            )
-            # FIXME UserWarning should not be raised here, but rather warned!
-            raise UserWarning(config_warning_string)
-        usage_warning_string = (
-            f'`{cls_name}` is not fully defined; you should define `{e.name}`, then call'
-            f' TODO! `methods.rebuild({cls_name})` before the first `{cls_name}` instance is created.'  # <--- TODO
-        )
-        cls.__pydantic_validator__ = MockValidator(  # type: ignore[assignment]
-            usage_warning_string, code='dataclass-not-fully-defined'
-        )
-        return False
```

Comment:
This is _all_ dead code. Since I'm removing `types_namespace` I removed it as well. @dmontagu / @samuelcolvin if either of you have a vision for this that I'm missing I'm happy to leave this in, but we should probably add some sort of test or exercise code this complex if left in.

Reply:
I think this stuff may eventually be necessary for recursive types and/or for “rebuilding” the dataclasses in the same scenarios you would for BaseModel. We haven’t implemented it yet, I don’t think it’s a huge deal if this code gets deleted though since we can re-add it as necessary. I think something similar will eventually show up here though.